app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt

```diff
@@ -80,6 +88,13 @@ open class NavigationActions(
     }
   }
 
+  open fun navigateAndClearStack(screen: String, clearUpToRoute: String) {
```

Comment:
Could you please explain why this function is needed? At the end of the creation process, we’ll be returning to a top-level destination, so I’m wondering why we can’t simply use `open fun navigateTo(destination: TopLevelDestination)`, which would fully clear the backstack. I may be overlooking something, so I’d appreciate any clarification you can provide.

Thank you!

Reply:
I implemented `navigateAndClearStack` specifically to return to `CreateRecipeScreen` after publishing a recipe, while clearing the backstack of previous screens. This keeps the user within the same top-level destination, allowing them to add another recipe if they choose. However, if the intended user flow is to navigate to a different screen (like a screen displaying the user’s created recipes) after publishing, then I agree my method might be redundant. Please let me know if I've misunderstood the user flow or if a different destination is intended after publishing!

Thank you for the feedback!

Reply:
@Ylanv I have deleted this method. So now, after publishing the recipe it will navigate to the SwipePage, that is why we dont need to clear the stack anymore.